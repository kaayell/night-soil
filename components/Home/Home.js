import React, {Component} from 'react'
import {Image, Text, View} from 'react-native'
import ActionButton from 'react-native-action-button'
import {BLUE, OFF_WHITE} from '../StyleGuide/colors'
import {POPPINS_MEDIUM} from '../StyleGuide/fonts'
import {Icon} from 'react-native-elements'
import style from '../StyleGuide/styles'
import Charts from '../Charts/Charts'

export class Home extends Component {
  static navigationOptions = ({navigation}) => ({
    title: <Image source={require('../../assets/poop.png')}
                  style={{width: 35, height: 35}}/>,
    headerStyle: {
      backgroundColor: BLUE,
    },
    headerTitleStyle: {
      color: OFF_WHITE,
      fontFamily: POPPINS_MEDIUM,
      alignSelf: 'center'
    },
    headerRight:
      <Icon name={'perm-identity'} iconStyle={style.icon} color={'white'}
            onPress={() => {
              navigation.navigate('Profile')
            }}
            underlayColor={BLUE}
            size={25}/>,

  })

  constructor(props) {
    super(props)

    this.state = {
      minutesToDate: 0,
      moneyToDate: 0,
    }
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <View style={{flex: 1}}>
          <Charts/>
        </View>
        <View style={{
          height: 100,
          flex: 1,
          backgroundColor: 'transparent',
        }}>
          <ActionButton position="right" buttonColor="rgb(154, 192, 205)">
            <ActionButton.Item buttonColor={BLUE} title="Record Poop"
                               onPress={() => this.props.navigation.navigate('Create')}>
              <Icon name="add"/>
            </ActionButton.Item>
            <ActionButton.Item buttonColor={BLUE} title="Time Poop"
                               onPress={() => this.props.navigation.navigate('Timer')}>
              <Icon name="hourglass-empty"/>
            </ActionButton.Item>
          </ActionButton>
        </View>
      </View>)
  }
}

export default Home