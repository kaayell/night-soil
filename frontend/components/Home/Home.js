import React, { Component } from 'react'
import { connect } from 'react-redux'
import * as apiClient from '../../api/apiClient'
import { Image, Text, View } from 'react-native'

export class Home extends Component {
  static navigationOptions = ({navigation}) => ({
    title: 'Home',
    tabBarIcon: ({tintColor}) =>
      <Image source={require('../../assets/icons/home_grey_24x24.png')}
             style={{tintColor: tintColor}}/>
  })

  constructor (props) {
    super(props)

    this.state = {
      minutesToDate: 0,
      moneyToDate: 0
    }
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.humanInfo) {
      apiClient.getSummary(nextProps.humanInfo.id)
        .then((response) => {
          if (response.data) {
            this.setState({
              minutesToDate: response.data.minutesToDate,
              moneyToDate: response.data.moneyToDate
            })
          }
        })
    }
  }

  render () {
    return <View><Text>Pooooopin</Text></View>
  }
}

const mapStateToProps = state => {
  return {
    humanInfo: state.humanInfo
  }
}

export default connect(mapStateToProps)(Home)