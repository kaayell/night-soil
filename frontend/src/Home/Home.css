.add-poop-button {
    position: fixed;
    bottom: 15%;
    right: 8%;
}