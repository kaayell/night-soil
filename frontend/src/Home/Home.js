import React, {Component} from 'react';
import {Card, CardText, CardTitle, FloatingActionButton} from "material-ui";
import {ContentAdd} from "material-ui/svg-icons/index"
import "./Home.css"
import {connect} from "react-redux";
import {setActivePage} from "../Navigation/navigation-actions";
import * as apiClient from "../api/apiClient"

export class Home extends Component {
    constructor(props) {
        super(props)

        this.state = {
            minutesToDate: 0,
            moneyToDate: 0
        }
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.humanInfo) {
            apiClient.getSummary(nextProps.humanInfo.id)
                .then((response) => {
                    if (response.data) {
                        this.setState({
                            minutesToDate: response.data.minutesToDate,
                            moneyToDate: response.data.moneyToDate
                        })
                    }
                })
        }
    }

    render() {
        return (
            <div className="home-container">
                <h2>Hi {this.props.humanInfo.firstName}</h2>
                <Card className="stat-card">
                    <CardTitle className="stat-header" title={"Minutes Pooping"}/>
                    <CardText className="stat">{this.state.minutesToDate}</CardText>
                </Card>
                <Card className="stat-card">
                    <CardTitle className="stat-header" title={"Money Made"}/>
                    <CardText className="stat">{this.state.moneyToDate}</CardText>
                </Card>
                <div className="add-poop-button">
                    <FloatingActionButton onClick={() => this.props.setActivePage("create")}>
                        <ContentAdd/>
                    </FloatingActionButton>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        humanInfo: state.humanInfo
    }
}

export default connect(mapStateToProps, {setActivePage})(Home)