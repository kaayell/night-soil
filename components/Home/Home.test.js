import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'

describe('Home', () => {

  it('should render', () => {
    let wrapper = shallow(<Home/>)
    expect(wrapper).toMatchSnapshot()
  });
})