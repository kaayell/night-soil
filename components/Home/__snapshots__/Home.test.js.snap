// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ChartsSummary />
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "justifyContent": "flex-end",
        "margin": 0,
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <Create
      closeModal={[Function]}
    />
  </ReactNativeModal>
  <ActionButton
    active={false}
    activeOpacity={0.85}
    autoInactive={true}
    backdrop={false}
    backgroundTappable={false}
    bgColor="transparent"
    bgOpacity={1}
    buttonColor="rgb(154, 192, 205)"
    buttonText="+"
    buttonTextStyle={Object {}}
    degrees={45}
    fixNativeFeedbackRadius={false}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    offsetX={30}
    offsetY={30}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOn={[Function]}
    outRangeScale={1}
    position="right"
    resetToken={null}
    size={56}
    spacing={20}
    useNativeFeedback={true}
    verticalOrientation="up"
  />
</View>
`;
